src/components/Button.tsx

```diff
@@ -176,3 +189,7 @@ const iconStyles: Record<ButtonSize, IconProps["xss"]> = {
 
 export type ButtonSize = "sm" | "md" | "lg";
 export type ButtonVariant = "primary" | "secondary" | "tertiary" | "danger" | "text";
+
+function isPromise(obj: void | Promise<void>): obj is Promise<void> {
+  return (typeof obj === "function" || typeof obj === "object") && "then" in obj && typeof obj.then === "function";
```

Comment:
Thought: I don't think the `typeof obj === function` is necessary? I.e. afaiu the "does `then` exist" is the canonical "is this a promise" check.